### DISCUSSION TITLE
How to change background image?

### DISCUSSION THREAD

**Discussion by thanglq1** [hm21/pro_image_editor#277] [Q&A]
I would like to change background image. How can I do it?
``` 
String _imagePath = "https://next-images.123rf.com/index/_next/image/?url=https://assets-cdn.123rf.com/index/static/assets/top-section-bg.jpeg&w=3840&q=75";

  @override
  Widget build(BuildContext context) {
    return ProImageEditor.network(
      _imagePath!,
      key: _editorKey,
      callbacks: _proImageEditorCallbacks,
      configs: _proImageEditorConfigs(constraints),
    );
  }
```

Now I have a button to change background image. I tried to setState but it does not work.

```
setState(() {
  _imagePath = "new image path";
  });
```

**Comment by hm21** [comment]
Before version `6.2.0`, replacing the background image required several steps. However, I just released version `6.2.0`, which simplify that part. You can now use the `updateBackgroundImage` method to update the background image. Simply call the method and pass your image as shown below:

```dart
  await editorKey.currentState!.updateBackgroundImage(
  EditorImage(
    networkUrl: 'https://picsum.photos/id/230/2000',
  ),
);
```

**Reply by thanglq1** [reply]
Thank you so much!

**Reply by yangheecho** [reply]
Thank you!!
